search_engine_parser/core/engines/baidu.py

```diff
@@ -0,0 +1,54 @@
+"""@desc 
+		Parser for Baidu search results
+"""
+
+from search_engine_parser.core.base import BaseSearch
+
+class BaiduSearch(BaseSearch):
+	"""
+	Searches Baidu for string
+	"""
+	name = "Baidu"
+	search_url = "https://www.baidu.com/s?wd={query}&pn={offset}&oq={query}"
+	summary = "\tBaidu, Inc. is a Chinese multinational technology company specializing in"\
+	" Internet-related services and products and artificial intelligence (AI), headquartered"\
+	" in Beijing's Haidian District.\n\tIt is one of the largest AI and internet"\
+	" companies in the world.\n\tBaidu offers various services, including a"\
+	" Chinese search engine, as well as a mapping service called Baidu Maps."
+
+	def parse_soup(self, soup):
+		"""
+		Parses Baidu for a search query
+		"""
+		
+		"""Baidu search can be made deterministic via an id, so add it in self"""
+		
+		soups=[]
+		for i in range(int(self.offset)*10+1,int(self.offset)*10+11):
```

Comment:
If baidu supports offset in urls, why try to offset the returned values in the for loop?

Reply:
The offsets on Baidu are like this: for page 1, the pn argument is 0, page 2, it's 10, page 3, it's 20 and so on. I was adding the field (page_number-1) in base.py and retrieving it here, so that I can get the id of the div's, as the ids are of form: 1-10 in page 1, 11-20 in page 2, etc...

Reply:
You can use regex expressions to select the div tags, you could use that instead. Don't you think?

Reply:
Thanks for the idea! Implemented as instructed 😄